core/templates/dev/head/pages/exploration_editor/statistics_tab/StatisticsTab.js

```diff
@@ -263,6 +263,9 @@ oppia.controller('StatisticsTab', [
                     visualizationsInfo[i].data));
                   el.attr('options', HtmlEscaperService.objToEscapedJson(
                     visualizationsInfo[i].options));
+                  el.attr('explorationId', $scope.explorationId);
+                  el.attr('state', HtmlEscaperService.objToEscapedJson(
```

Comment:
Can we just pass 'stateName' and retrieve the state from explorationStatesService? Passing the whole state object through JSON seems a bit heavy.

Reply:
I agree, but when I tried to use explorationStatesServices from the visualization.js I couldn't seem to get back the correct response (always received undefined instead..?) I'd also prefer to do it there but I can't figure out why I get back undefined.

Reply:
That's weird. Did you confirm that stateName was being passed in correctly? If so, then that should work...

Reply:
Yeah, it seems that stateName doesn't get passed properly but it's not entirely clear to me why... I've sent you a hangouts message w/ some info :(

Reply:
Done.